### PR TITLE
feat(torch): image data augmentation with random geometric perspectives

### PR DESCRIPTION
**Pull request by beniz**
tmp_1
tmp_5
tmp_12
tmp_1

API remains unchanged wrt. caffe backend.

This also fixes:
- cutout with BW images
- cutout probability setup via API

### CODE REVIEW

File: src/backends/torch/torchdataaug.cc [hunk 1]
```diff
@@ -125,8 +127,164 @@ namespace dd
...
       cv::Mat selected_area = src(rect);
-      cv::randu(selected_area, cv::Scalar(_cutout_vl, _cutout_vl, _cutout_vl),
-                cv::Scalar(_cutout_vh, _cutout_vh, _cutout_vh)); // TODO: bw
+      if (selected_area.channels() == 3)
+        cv::randu(selected_area,
+                  cv::Scalar(_cutout_vl, _cutout_vl, _cutout_vl),
+                  cv::Scalar(_cutout_vh, _cutout_vh, _cutout_vh));
+      else
+        cv::randu(selected_area,
+                  cv::Scalar(_cutout_vl, _cutout_vl, _cutout_vl),
```
Comment (Bycob) [inline src/backends/torch/torchdataaug.cc:136]: `cv::Scalar(_cutout_vl)`?

File: src/backends/torch/torchlib.cc [hunk 2]
```diff
@@ -604,9 +604,36 @@ namespace dd
...
+              geometry_persp_horizontal
+                  = ad_geometry.get("persp_horizontal").get<bool>();
+            if (ad_geometry.has("zoom_out"))
+              geometry_zoom_out = ad_geometry.get("zoom_out").get<bool>();
+            if (ad_geometry.has("zoom_in"))
+              geometry_zoom_in = ad_geometry.get("zoom_in").get<bool>();
+            if (ad_geometry.has("pad_mode"))
+              geometry_pad_mode = ad_geometry.get("pad_mode").get<int>();
+          }
+        std::cerr << "geometry=" << geometry_persp_vertical << std::endl;
```
Comment (Bycob) [inline src/backends/torch/torchlib.cc:631]: Remove? Or use logger?

**Review by Bycob — COMMENTED** [review]
Geometry function could be moved in a common location outside of torch, since they operate on cv images.
We can do it later after we add DTO.

File: src/backends/torch/torchlib.cc (hunk 2 above)
Comment (beniz) [reply src/backends/torch/torchlib.cc:631]: At the moment there's no access to the logger from the `TorchImgRandAugCV` object.

**Review by Bycob — APPROVED** [review]

**Review by fantes — APPROVED** [review]

**Pull request merged**